File: src/nexiss/schemas/analytics.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AnalyticsOverviewResponse(BaseModel):
    org_id: UUID
    total_documents_processed: int
    total_pages_processed: int
    total_llm_tokens_input: int
    total_llm_tokens_output: int
    processing_success_count: int
    processing_failed_count: int


class DailyProcessingPoint(BaseModel):
    day: datetime
    documents_processed: int
    pages_processed: int


class AnalyticsDailyProcessingResponse(BaseModel):
    org_id: UUID
    points: list[DailyProcessingPoint]

File: frontend/src/api/analytics.ts
import api from '@/lib/api'

export interface AnalyticsOverview {
  // Backend field names from AnalyticsOverviewResponse
  org_id: string
  total_documents_processed: number
  total_pages_processed: number
  total_llm_tokens_input: number
  total_llm_tokens_output: number
  processing_success_count: number
  processing_failed_count: number
}

export interface DailyPoint {
  // Backend field names from DailyProcessingPoint
  day: string              // was: date
  documents_processed: number  // was: documents
  pages_processed: number
}

export const analyticsApi = {
  overview: () => api.get<AnalyticsOverview>('/analytics/overview'),

  dailyProcessing: (days = 30) =>
    api.get<{ org_id: string; points: DailyPoint[] }>(
      `/analytics/daily-processing?days=${days}`
    ),

  usageSummary: () =>
    api.get<Array<{ metric_type: string; total: number }>>('/usage/summary'),
}
